File: src/main/java/com/example/demo/elevator/Elevator.java
package com.example.demo.elevator;

import lombok.Data;

@Data
public class Elevator {

    private final PositionManager pm;
    private final RunningSystem rs;
    private final RequestPool rp;

    public Elevator() {
        this.pm = new PositionManager();
        this.rp = new RequestPool(pm);
        this.rs = new RunningSystem();
        init();
    }

    private void init() {
        pm.setRunningSystem(rs);
        rs.setPositionManager(pm);
    }

    public void start() {
        rs.doRun();
    }

    public Connection connect() {
        return new Connection(this.rp);
    }
}

File: src/Elevator.js
import { Constants } from './Constants.js';
import { RequestPool } from './RequestPool.js';
import { PositionManager } from './PositionManager.js';
import { RunningSystem } from './RunningSystem.js';

class Elevator {
    constructor(eventBus) {
        this.eventBus = eventBus;
        this.rp       = new RequestPool(eventBus);
        this.pm       = new PositionManager(eventBus);
        this.rs       = new RunningSystem(eventBus);

        init();
    }

    init() {
       this.pm.setRunningSystem(this.rs);
       this.rs.setPositionManager(this.pm);
    }

    start() {
        // 用nodejs开启一个全新的线程
        this.rs.doRun();
    }

}